### PR TITLE
Always have l1t::EtSum::type_ be initialized

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Added new `kUninitialized` value to enum and used in default constructor
- Added new constructor to guarantee type is set
- Used new constructor in all places it seemed to make sense

This was found by the UBSAN checker.

#### PR validation:

Code compiles. This should not change any behavior (unless the behavior was caused by undefined behavior) of the code.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
In general, using a default constructor and then calling a bunch of `set*` calls is not a good idea. What really should be done is to add new constructors to `EtSum` to handle the actual cases being used to build an `EtSum` in this code.  But such work was more than I wanted to do at the time to solve the underlying issue.

**Comment by cecilecaillol** [comment]
+l1

File: EventFilter/L1TRawToDigi/plugins/implementations_stage1/EtSumUnpacker.cc [hunk 1]
```diff
@@ -39,16 +39,15 @@ namespace l1t {
...
         int flagtotet = et.hwQual();
         flagtotet |= overflowtotet;
         et.setHwQual(flagtotet);
         LogDebug("L1T") << "ET: pT " << et.hwPt() << "is overflow " << overflowtotet << std::endl;
         res_->set(bx, 2, et);
 
-        l1t::EtSum ht = l1t::EtSum();
+        l1t::EtSum ht{l1t::EtSum::kTotalHt};
         ht.setHwPt(totht);
         ht.setType(l1t::EtSum::kTotalHt);
```
Comment (perrotta) [inline EventFilter/L1TRawToDigi/plugins/implementations_stage1/EtSumUnpacker.cc:52]: This could be removed now, isn't it?

Comment (Dr15Jones) [reply EventFilter/L1TRawToDigi/plugins/implementations_stage1/EtSumUnpacker.cc]: Good catch. Thanks!

**Comment by Dr15Jones** [comment]
please test

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge
(@cecilecaillol I keep your previous signature for L1 for good: the last update on top of it was really trivial)

**Pull request merged**